### PR TITLE
feat: do not suggest collecting if available funds < basic output min storage deposit or there arent enough outputs to consolidate

### LINKED ISSUE #7519: [Task]: Do not suggest collecting if available funds < basic output min storage deposit or there arent enough outputs to consolidate
### Task description

We shouldnt motivate the users to collect their vesting outputs if they cant cover the storage deposit, as it will fail to collect. 
Do not suggest collecting if available funds < basic output min storage deposit

Same for outputs, the sdk would not allow consolidating if there arent enough available outputs, we should check if the sdk provides us with enough tools so we can check if there are enough outputs that can be used in the input selection algorithm

### Requirements

Do not suggest collecting if available funds < basic output min storage deposit

### Creation checklist

- [x] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

Closes #7519 
...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 1]
```diff
@@ -31,8 +35,14 @@
         MeatballMenuButton,
         TooltipIcon,
     } from '@ui'
+    import { onMount } from 'svelte'
 
     const DEFAULT_EMPTY_VALUE_STRING = '----'
+    let modal: Modal
+    let timeUntilNextPayout = DEFAULT_EMPTY_VALUE_STRING
+    let minRequiredStorageDeposit: number
+    const availableOuputs = undefined
```
Comment (cpl121) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:44]: this variable is always `undefined`. I think that before you updated it in `areOutputsReadyForConsolidation`, because the boolean that the promise returns is not being used

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 2]
```diff
@@ -56,11 +66,28 @@
...
+        areOutputsReadyForConsolidation()
+    })
+
+    async function areOutputsReadyForConsolidation(): Promise<boolean> {
+        let hasOutputsToConsolidate = false
+        try {
+            await $selectedAccount.prepareConsolidateOutputs({ force: false, outputThreshold: 2 })
+            hasOutputsToConsolidate = true
+        } catch (error) {
+            console.error('error ', error)
```
Comment (cpl121) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:82]: It is not necessary to show this error I think

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 3]
```diff
@@ -56,11 +66,28 @@
...
-    let modal: Modal
-    let timeUntilNextPayout = DEFAULT_EMPTY_VALUE_STRING
     $: $selectedAccountVestingPayouts, (timeUntilNextPayout = getTimeUntilNextPayout())
 
+    onMount(() => {
+        getMinStorage()
+        areOutputsReadyForConsolidation()
+    })
+
+    async function areOutputsReadyForConsolidation(): Promise<boolean> {
```
Comment (cpl121) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:76]: I think that this function, instead of returning a boolean, could change the value of a variable in a `then`
Something like that
```suggestion
    let hasOutputsToConsolidate: boolean =  false
    async function areOutputsReadyForConsolidation(): Promise<void> {
        $selectedAccount.prepareConsolidateOutputs({ force: false, outputThreshold: 2 })
            .then(() => hasOutputsToConsolidate = true)
            .catch(() => hasOutputsToConsolidate = false)
    }
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 4]
```diff
@@ -56,11 +66,24 @@
...
+    })
+
+    function areOutputsReadyForConsolidation(): void {
+        $selectedAccount
+            .prepareConsolidateOutputs({ force: false, outputThreshold: 2 })
+            .then(() => (hasOutputsToConsolidate = true))
+            .catch(() => (hasOutputsToConsolidate = false))
+    }
+
+    async function getMinRequiredStorageDeposit() {
```
Comment (cpl121) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:83]: same as `getMinRequiredStorageDeposit`, I think it would be better to modify the variable inside a `then` instead of making the function asynchronous

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by cpl121 — APPROVED** [review]
Perfect! 🚀

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 5]
```diff
@@ -56,11 +66,26 @@
...
         },
     ]
-
-    let modal: Modal
-    let timeUntilNextPayout = DEFAULT_EMPTY_VALUE_STRING
     $: $selectedAccountVestingPayouts, (timeUntilNextPayout = getTimeUntilNextPayout())
 
+    onMount(() => {
+        getMinRequiredStorageDeposit()
+        areOutputsReadyForConsolidation()
```
Comment (begonaalvarezd) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:73]: I think we should make `areOutputsReadyForConsolidation` reactive on wallet changes, if you receive a new output or something it should automatically update

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 6]
```diff
@@ -56,11 +66,26 @@
...
+        $selectedAccount
+            .prepareConsolidateOutputs({ force: false, outputThreshold: 2 })
+            .then(() => (hasOutputsToConsolidate = true))
+            .catch(() => (hasOutputsToConsolidate = false))
+    }
+
+    function getMinRequiredStorageDeposit() {
+        getRequiredStorageDepositForMinimalBasicOutput()
+            .then((deposit) => (minRequiredStorageDeposit = deposit))
+            .catch(() => (minRequiredStorageDeposit = 0))
```
Comment (begonaalvarezd) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:86]: catching an error and using 0 as the value is going to give problems because you have `$selectedAccount?.balances?.baseCoin?.available < minRequiredStorageDeposit` to disable the button, if this is 0 and I have 1micro, the condition would not be right, I feel like we shouldnt no allow collecting if you cant get the min storage deposit

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 7]
```diff
@@ -31,8 +35,14 @@
         MeatballMenuButton,
         TooltipIcon,
     } from '@ui'
+    import { onMount } from 'svelte'
 
     const DEFAULT_EMPTY_VALUE_STRING = '----'
+    let modal: Modal
+    let timeUntilNextPayout = DEFAULT_EMPTY_VALUE_STRING
+    let minRequiredStorageDeposit: number
+    let hasOutputsToConsolidate = false
```
Comment (begonaalvarezd) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:44]: I would create a variable called "canCollect" or something like that which has the logic in a reactive way, so we dont have to add the logic in the html section, we now have `{$selectedAccountVestingUnclaimedFunds === 0 ||
                                        hasTransactionInProgress ||
                                        $selectedAccount?.balances?.baseCoin?.available < minRequiredStorageDeposit ||
                                        hasOutputsToConsolidate === false` which is getting a bit out of hand imo 🙏🏼

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: packages/desktop/views/dashboard/vesting/Vesting.svelte [hunk 8]
```diff
@@ -56,10 +66,33 @@
...
-
-    let modal: Modal
-    let timeUntilNextPayout = DEFAULT_EMPTY_VALUE_STRING
     $: $selectedAccountVestingPayouts, (timeUntilNextPayout = getTimeUntilNextPayout())
+    $: canCollect =
+        $selectedAccountVestingUnclaimedFunds === 0 ||
+        hasTransactionInProgress ||
+        (minRequiredStorageDeposit !== null &&
+            $selectedAccount?.balances?.baseCoin?.available < minRequiredStorageDeposit) ||
+        hasOutputsToConsolidate === false
```
Comment (brancoder) [inline packages/desktop/views/dashboard/vesting/Vesting.svelte:75]: 1. Can we change this line to just `hasOutputsToConsolidate`, instead of:
       
 ```
hasOutputsToConsolidate === false
            ? false
            : true
``````
2. These conditions and naming `canCollect` are a bit confusing:

`canCollect` should be `true` if :

- have funds to claim: $selectedAccountVestingUnclaimedFunds > 0
- there are no transactions in progress so: `!hasTransactionInProgress`
- if you have enough balance available to cover minRSD =>  `$selectedAccount?.balances?.baseCoin?.available > minRequiredStorageDeposit`
- and `hasOutputsToConsolidate === true` 

So to sum it up you can collect if there are funds to claim, no transactions in progress, have available funds to cover MRSD  and have outputs to consolidate.

Maybe I have misunderstood this :see_no_evil:

**Review by brancoder — CHANGES_REQUESTED** [review]

**Review by brancoder — CHANGES_REQUESTED** [review]
There seams to be an issue with `areOutputsReadyForConsolidation`.
If user navigates back and forward(without collecting) to Vesting tab first time `prepareConsolidateOutputs`  returns true and second time it returns an error that there are no outputs to consolidate and Collect button is disabled.
[Screencast from 25.10.2023 10:17:02.webm](https://github.com/iotaledger/firefly/assets/52735957/3ece3520-d132-488a-99fd-5f3e461bc5d6)

**Review by begonaalvarezd — APPROVED** [review]

**Review by cpl121 — APPROVED** [review]
NJ

**Review by brancoder — APPROVED** [review]
LGTM
